lib/restrictions.js

```diff
@@ -17,7 +20,11 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
   await promisify(database.createIndexes)([{
     // for getting restrictions for a particular resource within a zone
     collection: 'resource-restriction-restriction',
-    fields: {'restriction.resource': 1, 'restriction.zone': 1},
+    fields: {
+      'restriction.resource': 1,
+      'restriction.zone': 1,
+      'restriction.methodOptions.duration': 1
```

Comment:
The previous model was such that there could only be one restriction per resource per zone. If we're going to allow multiple restrictions per zone, then we should give restrictions IDs and check uniqueness on that basis, not based on some other parameters associated with a particular type of restriction.

Reply:
I have updated the code to use an id that is a hash of the zone, resource, methodOptions, and if there is a duration, the duration.

Reply:
See my comment on the hash code -- it doesn't achieve the goal in my comment here, which is to not check for uniqueness based on restriction parameters, but only based on a restriction identifier. If that restriction identifier just becomes a deterministic result of the restriction parameters, then we're checking for uniqueness based on restriction parameters when we check the ID -- just indirectly. My understanding is that we've decided that you can have any number of restrictions you want using any parameters you want -- repeated or not.

It's up to the maintainer of restrictions to decide what restrictions exist, we don't try and guess/help them not make duplicates according to some (too) limited set of information that we possess at the low level. Remember that they can create all manner of new restriction types in their extensions -- and if we wanted to try and enforce uniqueness on that basis we'd need to expose an API to allow the extensions to produce their own unique ID. Ultimately, the caller should be able to just set their own restriction ID -- and they can do whatever they want with that field (make them random or deterministic based on whatever criteria they please).